frontend/src/components/samples/details/SampleDetailsLineage.js

```diff
@@ -202,10 +201,19 @@ const SampleDetailsLineage = ({
                     id="graph-id"
                     data={adjustedGraphData}
                     config={graphConfig}
-                    onClickNode={(id, _) => history(`/samples/${id}`)}
+                    // tabPaneKey is used to generate a hash url for the lineage tab.
+                    // Without the hash url, clicking a node navigates the user to the Overview tab
+                    // rather than staying in the graph.
+                    onClickNode={(id, _) => {
+                      if (id && handleSampleClick) {
```

Comment:
is handleSampleClick ever false ? maybe skip this test and remove the if using id && handleSampleClick(id)

Reply:
I am not sure we want failures to be silent (it might be partially desirable in some case but here if we forget to set a callback i would expect a failure) if we want to remove the functionality we just remove the hook altogether. At the moment, it is not optional. Since we provide both the hook and the callback i would take a missing callback as a programming error which we typically don't test for. If you want to keep the test then add an else so it actually fails when a callback is not provided (or at least a console message is present to make debugging easier).

Reply:
It's a standard pattern in react for a component to take callback functions as props, and to make those props optional. The caller can choose to pass a function, or leave it undefined. If the component receives a callback it will call it. If not, it ignores it gracefully, without exploding. This is the expected behaviour for callback props. There's nothing in the lineage graph that requires either of these two callbacks to be defined, so they are optional. 
I don't agree with the idea of purposefully causing an error that results in a blank page for the user as a way of guarding against a programming error. I'd much rather have the user report that nothing is happening when they click a node than have the app crash.

Reply:
We made that decision already we don't want those optional ... we want them to link to sample and processMeasurement. The optional test is purposeful testing of programming error since the only way this could be true is if we forget to add the callbacks ... making the choice of making something optional is not driven by the component, but by us. If we start leaving all those "options" open this will cause much more harm than good. Easiest way to see it is each time you can remove an if do it ...